src/botPage/bot/TradeEngine/Ticks.js

```diff
@@ -31,18 +31,33 @@ export default Engine =>
                 tickListenerKey = key;
             }
         }
-        getTicks() {
+        getTicks(toString = false) {
             return new Promise(resolve =>
-                this.$scope.ticksService
-                    .request({ symbol: this.symbol })
-                    .then(ticks => resolve(ticks.map(o => o.quote)))
+                this.$scope.ticksService.request({ symbol: this.symbol }).then(ticks =>
```

Comment:
- To make the code more readable we can separate some of the calls here, e.g. by assigning values to variables instead of nesting everything.
- We can store `this.getPipSize()` in a variable so we don't need to call that function in each `map` iteration.

```JavaScript
this.$scope.ticksService
    .request({ symbol: this.symbol })
    .then(ticks => {
        const pipSize = this.getPipSize();
        const ticksList = ticks.map(tickObj => {
            if (toString) {
                return tickObj.quote.toFixed(pipSize);
            }
            return tickObj.quote;
        });

        resolve(ticksList);
    })
```